algebird-core/src/main/scala/com/twitter/algebird/SpaceSaver.scala

```diff
@@ -101,14 +107,14 @@ object SSMany {
   private def bucketsFromCounters[T](counters: Map[T, (Long, Long)]): SortedMap[Long, Set[T]] =
     SortedMap[Long, Set[T]]() ++ counters.groupBy(_._2._1).mapValues(_.keySet)
 
-  private def apply[T](capacity: Int, counters: Map[T, (Long, Long)]): SSMany[T] =
+  private[algebird] def apply[T](capacity: Int, counters: Map[T, (Long, Long)]): SSMany[T] =
     SSMany(capacity, counters, bucketsFromCounters(counters))
 
   private[algebird] def apply[T](one: SSOne[T]): SSMany[T] =
     SSMany(one.capacity, Map(one.item -> (1L, 0L)), SortedMap(1L -> Set(one.item)))
 }
 
-case class SSMany[T](capacity: Int, counters: Map[T, (Long, Long)], buckets: SortedMap[Long, Set[T]]) extends SpaceSaver[T] {
+case class SSMany[T] private (capacity: Int, counters: Map[T, (Long, Long)], buckets: SortedMap[Long, Set[T]]) extends SpaceSaver[T] {
```

Comment:
the  2 apply methods in companion object are now both called from other classes so they both need to be private[algebird]. the one on line 110 had to be opened up because of new SpaceSaver.apply method on line 16.

the main constructor is not used outside the class. it can also very easily be used incorrectly (e.g. create buckets that are inconsistent with counters), so i rather keep it as private as possible.